### PR TITLE
ENH Add generic types

### PR DESCRIPTION
**Pull request by GuySartorelli**
Requires https://github.com/silverstripe/silverstripe-framework/pull/11108 for these generics to take effect.

## Issue
- https://github.com/silverstripe/silverstripe-framework/issues/11066

### CODE REVIEW

File: src/Extension/Publishable/PublishableSiteTree.php [hunk 1]
```diff
@@ -36,7 +38,7 @@ public function getMyVirtualPages()
     }
 
     /**
-     * @return array|SS_List
+     * @return iterable<SiteTree>
```
Comment (emteknetnz) [inline src/Extension/Publishable/PublishableSiteTree.php:41]: Also update StaticPublishingTrigger return type

File: src/Extension/Publishable/PublishableSiteTree.php [hunk 2]
```diff
@@ -92,7 +94,7 @@ public function objectsToUpdate($context)
     /**
      * This method controls which caches will be purged
      *
-     * @return array|SS_List
+     * @return iterable<SiteTree>
```
Comment (emteknetnz) [inline src/Extension/Publishable/PublishableSiteTree.php:97]: Also update StaticPublishingTrigger return type

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: src/Extension/Publishable/PublishableSiteTree.php (hunk 1 above)
Comment (ChloeSartorelli) [reply src/Extension/Publishable/PublishableSiteTree.php:41]: Good catch. Done

File: src/Extension/Publishable/PublishableSiteTree.php (hunk 2 above)
Comment (ChloeSartorelli) [reply src/Extension/Publishable/PublishableSiteTree.php:97]: Done

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**